test/e2e_node/garbage_collector_test.go

```diff
@@ -257,7 +257,10 @@ func containerGCTest(f *framework.Framework, test testRun) {
 
 // Runs containerGCTest using the docker runtime.
 func dockerContainerGCTest(f *framework.Framework, test testRun) {
-	runtime := docker.ConnectToDockerOrDie(defaultDockerEndpoint, defaultRuntimeRequestTimeoutDuration)
+	var runtime docker.DockerInterface
+	BeforeEach(func() {
+		runtime = docker.ConnectToDockerOrDie(defaultDockerEndpoint, defaultRuntimeRequestTimeoutDuration)
```

Comment:
@timstclair I see.

When run in parallel, ginkgo starts multiple **processes** to run all the test.

In each process, tests are running in serial. So:
* if 2 tests are running in parallel, they won't share anything because they are in different processes.
* If 2 tests are in the same process, they won't share anything at the same time because they run one after another.